core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/SearchModelImpl.java

```diff
@@ -151,6 +164,24 @@ private void init() {
         configJson = getSearchConfigJson();
     }
 
+    public ConnectionFailedAlert resolveConnectionFailedAlert() {
+        SearchCAConfigurationModel searchCAConfigurationModel = resource.adaptTo(SearchCAConfigurationModel.class);
+        if (searchCAConfigurationModel == null) {
+            log.warn("Can not resolve context aware search configuration model.");
+            return new ConnectionFailedAlert(AlertVariant.WARNING, Collections.singletonList(i18n.get(I18N_SEARCH_CA_CONFIGURATION_FAILED_TO_RESOLVE, null, resource.getPath())));
+        }
+        try {
+            boolean pingSuccess = fulltextSearchPingService.ping(searchCAConfigurationModel);
+            if (!pingSuccess) {
+                return new ConnectionFailedAlert(AlertVariant.ERROR, Arrays.asList(i18n.get(I18N_SEARCH_CONNECTION_FAILED_FURTHER_ACTION_CHECK_OSGI_CONFIGURATION), i18n.get(I18N_SEARCH_CONNECTION_FAILED_FURTHER_ACTION_CHECK_LOG_FILES)));
+            }
+        } catch (Exception e) {
+            return new ConnectionFailedAlert(AlertVariant.WARNING, Collections.singletonList(e.getMessage()));
+        }
+        return null;
```

Comment:
this will only happn in case of the exception block, no? in that case you can change to 
`             return new ConnectionFailedAlert(AlertVariant.WARNING, Collections.singletonList(e.getMessage()));`

Reply:
I need the returning of 'null' for the case when connection is established

Reply:
aha ok I see yes